app/controllers/comments_controller.rb

```diff
@@ -110,6 +110,35 @@ def create
     render json: { error: message }, status: :unprocessable_entity
   end
 
+  def moderator_create
+    return if RateLimitChecker.new(current_user).limit_by_action("comment_creation")
+
+    canned_response = CannedResponse.find(params[:canned_response][:id])
+    authorize canned_response, :moderator_create?
+
+    moderator = User.find(SiteConfig.user_moderator_id)
+    @comment = Comment.new(permitted_attributes(Comment))
+    @comment.user_id = moderator.id
+    @comment.body_markdown = canned_response.content
+    @comment.actor_id = current_user.id
+    authorize @comment
+
+    if @comment.save
+      Mention.create_all(@comment)
+      Notification.send_new_comment_notifications_without_delay(@comment)
+
+      render json: { status: "created" }
+      # ask ben about this elsif
+    elsif (@comment = Comment.where(body_markdown: @comment.body_markdown,
+                                    commentable_id: @comment.commentable.id,
+                                    ancestry: @comment.ancestry)[1])
+      @comment.destroy
```

Comment:
Re: conversation with Ben, the whole `elsif` clause is to prevent duplicate comments if a user were to hit the submit button twice really quickly or something similar.

Reply:
I think we should add that info in a comment before the elsif, so that I won't ask you again in 5 days :D